packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -89,10 +133,17 @@ class TextEditingController extends ValueNotifier<TextEditingValue> {
   /// that they need to update (it calls [notifyListeners]). For this reason,
   /// this value should only be set between frames, e.g. in response to user
   /// actions, not during the build, layout, or paint phases.
+  ///
+  /// This property can be set from a listener added to this
+  /// [TextEditingController] however one should not also set [selection].
```

Comment:
I've rewritten this statement a little differently. I was wary of emphasizing the "same time" since that's in fact the goal.  However, I think the revised version is a little clearer.

Reply:
Yeah, that's a lot better.